File: task4_java/Wand.java
public class Wand extends Weapon{
	private static final int WAND_RANGE = 5;
	private static final int WAND_INIT_HEAL = 10;

	public Wand(Player owner)
	{
		super(WAND_RANGE, WAND_INIT_HEAL, owner);
	}


	public void enhance(){

		this.effect += 5;
	}

	@Override
	public void action(int posx, int posy){

		System.out.println("You are using wand to heal " + posx + " " + posy + ".");

		if (this.owner.pos.distance(posx, posy)  <= this.range) {
			// search for all targets with target coordinates.
			Player player = owner.game.getPlayer(posx, posy);

			if(player != null ) 
			{
				if(player.getClass().equals( owner.getClass())){
					player.increaseHealth(this.effect);
					System.out.println("You have healed " + player.myString ); 
				}
				else {
					System.out.println(" The target you're healing is not of the same race");
				}

				
			}
		} else {
			System.out.println("Out of reach.");
		}
	}

}

File: task4_python/Wand.py
from Weapon import Weapon

class Wand(Weapon):
	WAND_RANGE = 5
	WAND_INIT_HEAL = 10

	def __init__(self,owner):
		super(Wand, self).__init__(Wand.WAND_RANGE, Wand.WAND_INIT_HEAL, owner)

	def enhance(self):
		self.effect +=5

	def action(self, posx, posy):
		print"You are using wand healing %d %d." %(posx,posy)

		if self.owner.pos.distance1(posx,posy)<= self.ranges:
			player = self.owner.game.getPlayer(posx,posy)
			
			if player != None :
		
				
				if type(player)is type(self.owner):
					
					player.increaseHealth(self.effect)
					print "You have healed %s for %d" %(player.myString , self.effect)
				else:
					print"The target you're healing is not of the same race"
				
		else:
			print"Out of reach."

